src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -134,7 +134,7 @@ private[impl] class KillServiceActor(
     logger.debug(s"Adding instances $instanceIds to the queue")
     maybePromise.map(p => p.completeWith(KillStreamWatcher.watchForKilledTasks(instanceTracker.instanceUpdates, instances).runWith(Sink.ignore)))
     instances
-      .filterNot(instance => inFlight.keySet.contains(instance.instanceId) && instance.tasksMap.nonEmpty) // Don't trigger a kill request for instances that are already being killed
+      .filterNot(instance => inFlight.keySet.contains(instance.instanceId) || instance.tasksMap.isEmpty) // Don't trigger a kill request for instances that are already being killed
```

Comment:
oh boy...

Reply:
Gosh. I checked it twice on the original review and did not see an issue 🙈